frontend/src/component/pages/HomePage.tsx

```diff
@@ -107,16 +108,20 @@ export class HomePage extends Component<Props, State> {
     });
     this.handleStateChange();
     this.autoFillLongLink();
+    try {
+      const changeLog = await this.props.changeLogService.getChangeLog();
+      this.setState({ changeLog: changeLog.changes }, async () => {
+        const hasUpdates = await this.props.changeLogService.hasUpdates();
+        if (!hasUpdates) {
+          return;
+        }
 
-    const changeLog = await this.props.changeLogService.getChangeLog();
-    this.setState({ changeLog }, async () => {
-      const hasUpdates = await this.props.changeLogService.hasUpdates();
-      if (!hasUpdates) {
-        return;
-      }
-
-      this.showChangeLogs();
-    });
+        this.showChangeLogs();
+      });
+    } catch (err) {
+      const { changeLogErr } = err;
+      this.props.store.dispatch(raiseGetChangeLogError(changeLogErr));
+    }
```

Comment:
Do we want to load change log when component is loading or wait until user tries to open the modal? I recommend we wrap this into separate function: `autoShowChangeLog`, `autoShowChangeLog`, or something else to reduce complexity.